platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/result/DefaultResolvedComponentResult.java

```diff
@@ -179,4 +198,8 @@ public static void eachElement(
             }
         }
     }
+
+    public void complete() {
+        dependents = ImmutableSet.copyOf(dependents);
+    }
```

Comment:
🤔 I think this needs a short explanation.  Why should it be called?  What is "completed"?  Why are we putting the responsibility for "finalizing" the collection as an immutable type onto clients of this class, especially if they can only get an unmodifiable set view of this collection anyways.